File: implementations/java-springboot/examples/src/main/java/com/example/accountservice/account/domain/MoneyWithdrawnEvent.java
package com.example.accountservice.account.domain;

import java.time.LocalDateTime;

public record MoneyWithdrawnEvent(
        String accountId,
        String email,
        String transactionId,
        Money amount,
        Money balanceAfter,
        // Carried through so CategorizeTransactionEventHandler doesn't need a separate lookup to
        // react — the same reasoning as every other field on this event. Absent when the requester
        // didn't attach one; CategorizeTransactionEventHandler skips categorization entirely in
        // that case.
        String merchantName,
        LocalDateTime createdAt) {}

File: implementations/kotlin-springboot/examples/src/main/kotlin/com/example/accountservice/account/domain/MoneyWithdrawnEvent.kt
package com.example.accountservice.account.domain

import java.time.LocalDateTime

data class MoneyWithdrawnEvent(
    override val accountId: String,
    override val email: String,
    val transactionId: String,
    val amount: Money,
    val balanceAfter: Money,
    val createdAt: LocalDateTime,
    // Carried through so CategorizeTransactionEventHandler doesn't need a separate lookup to react —
    // the same reasoning as every other field on this event. Absent when the requester didn't attach
    // one; CategorizeTransactionEventHandler skips categorization entirely in that case.
    val merchantName: String? = null,
) : DomainEvent
